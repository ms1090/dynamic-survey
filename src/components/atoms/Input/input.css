.text-input {
  display: block;
}
.text-input input {
  border-radius: 3px;
  border: solid 1px #8c8c8a;
  box-sizing: border-box;
  color: #5d5d5d;
  display: block;
  height: 40px;
  margin: 10px 0;
  padding: 10px 16px;
  width: 100%;
}