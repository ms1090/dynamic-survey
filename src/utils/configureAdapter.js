import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
// configure adapter to use enzyme in unit test cases
configure({ adapter: new Adapter() });
