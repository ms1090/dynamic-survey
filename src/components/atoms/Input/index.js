import React from 'react';
import PropTypes from 'prop-types';
import './input.css';

/**
 * Display input element as per customization props
 * @param {String} className CSS class name
 * @param {String} id Input element id
 * @param {String} label Label for input element
 * @return {React.Element} Return input element
 */
const Input = ({
  className,
  id,
  label,
  ...others
}) => (
  <label className={`text-input ${className}`} htmlFor={id}>
    <span>{label}</span>
    <input
      id={id}
      {...others}
    />
  </label>
);
// default values for component props
Input.defaultProps = {
  className: '',
  id: '',
  label: 'Form input',
};
// prop types checking
Input.propTypes = {
  className: PropTypes.string,
  id: PropTypes.string,
  label: PropTypes.string,
};
export default Input;
