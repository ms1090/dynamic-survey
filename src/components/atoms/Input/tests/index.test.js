import React from 'react';
import { shallow } from 'enzyme';
import Input from '..';
import toJson from 'enzyme-to-json';
import 'utils/configureAdapter';

describe('Input Component', () => {
  test('should render correctly', () => {
    const wrapper = shallow(<Input
      id="testId"
      className="test-className"
      label="test label"
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
