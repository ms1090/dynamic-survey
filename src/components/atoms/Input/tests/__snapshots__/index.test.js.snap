// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component should render correctly 1`] = `
<label
  className="text-input test-className"
  htmlFor="testId"
>
  <span>
    test label
  </span>
  <input
    id="testId"
  />
</label>
`;
